src/client/datascience/telemetry/interpreterPackages.ts

```diff
@@ -0,0 +1,109 @@
+);
+
+@injectable()
+export class InterpreterPackages {
+    private static interpreterInformation = new Map<string, Map<string, string>>();
+    private static pendingInterpreterInformation = new Map<string, Promise<void>>();
+    constructor(
+        @inject(IPythonExtensionChecker) private readonly pythonExtensionChecker: IPythonExtensionChecker,
+        @inject(IInterpreterService) private readonly interpreterService: IInterpreterService,
+        @inject(IPythonExecutionFactory) private readonly executionFactory: IPythonExecutionFactory
+    ) {}
+    public static getPackageVersions(interpreter: PythonEnvironment): Map<string, string> | undefined {
+        return InterpreterPackages.interpreterInformation.get(interpreter.path);
+    }
+    public trackPackages(interpreterUri: InterpreterUri, ignoreCache?: boolean) {
+        this.trackPackagesInternal(interpreterUri, ignoreCache).catch(noop);
+    }
+    public async trackPackagesInternal(interpreterUri: InterpreterUri, ignoreCache?: boolean) {
+        if (!this.pythonExtensionChecker.isPythonExtensionInstalled) {
+            return;
+        }
+        let interpreter: PythonEnvironment;
+        if (isResource(interpreterUri)) {
+            // Get details of active interpreter for the Uri provided.
+            const activeInterpreter = await this.interpreterService.getActiveInterpreter(interpreterUri);
+            if (!activeInterpreter) {
+                return;
+            }
+            interpreter = activeInterpreter;
+        } else {
+            interpreter = interpreterUri;
+        }
+        this.trackInterpreterPackages(interpreter, ignoreCache).catch(noop);
+    }
+    private async trackInterpreterPackages(interpreter: PythonEnvironment, ignoreCache?: boolean) {
+        const key = interpreter.path;
+        if (InterpreterPackages.pendingInterpreterInformation.has(key) && !ignoreCache) {
+            return;
+        }
+
+        const promise = this.getPackageInformation(interpreter);
+        promise.finally(() => {
+            // If this promise was resolved, then remove it from the pending list.
+            if (InterpreterPackages.pendingInterpreterInformation.get(key) === promise) {
+                InterpreterPackages.pendingInterpreterInformation.delete(key);
+            }
+        });
+        InterpreterPackages.pendingInterpreterInformation.set(key, promise);
+    }
+    @captureTelemetry(Telemetry.AttemptedToLaunchRawKernelWithoutInterpreter)
+    private async getPackageInformation(interpreter: PythonEnvironment) {
+        const service = await this.executionFactory.createActivatedEnvironment({
+            allowEnvironmentFetchExceptions: true,
+            bypassCondaExecution: true,
+            interpreter
+        });
+
+        // Ignore errors, and merge the two (in case some versions of python write to stderr).
+        const output = await service.execModule('pip', ['list'], { throwOnStdErr: false, mergeStdOutErr: true });
```

Comment:
Does pip always work in a conda situation?

Reply:
Yes, also we have no way of invoking conda commands, Python extension has that API & we don't.
But i checked this & it works well enough for us.

Reply:
I.e. pip will have to do for now.